src/components/SearchBar.tsx

```diff
@@ -15,14 +15,28 @@ type SearchBarProps = {
 
 function SearchBar({ queryParams, size }: SearchBarProps) {
   const [query, setQuery] = useState("");
+  const [suggestions, setSuggestions] = useState<string[]>([]);
   const router = useRouter();
 
   let sizeClass = "h-11";
-
   if (size === "large") {
     sizeClass = "h-14";
   }
 
+  useEffect(() => {
+    if (query.trim()) {
```

Comment:
Also maybe try to add some throttling here because right now it is making request on every keypress which is not ideal.

You can read more here
https://www.telerik.com/blogs/debouncing-and-throttling-in-javascript